nixos/modules/services/continuous-integration/buildkite-agents.nix

```diff
@@ -215,52 +232,55 @@ in
+    mkMerge ((flip map) (range 1 cfg.count) (n: {
+      "buildkite-agent-${name}-${toString n}" =
+        { description = "Buildkite Agent";
+          wantedBy = [ "multi-user.target" ];
+          after = [ "network.target" ];
+          path = cfg.runtimePackages ++ [ cfg.package pkgs.coreutils ];
+          environment = config.networking.proxy.envVars // {
+            HOME = cfg.dataDir;
+            NIX_REMOTE = "daemon";
+          };
 
-        ## NB: maximum care is taken so that secrets (ssh keys and the CI token)
-        ##     don't end up in the Nix store.
-        preStart = let
-          sshDir = "${cfg.dataDir}/.ssh";
-          tagStr = lib.concatStringsSep "," (lib.mapAttrsToList (name: value: "${name}=${value}") cfg.tags);
-        in
-          optionalString (cfg.privateSshKeyPath != null) ''
-            mkdir -m 0700 -p "${sshDir}"
-            cp -f "${toString cfg.privateSshKeyPath}" "${sshDir}/id_rsa"
-            chmod 600 "${sshDir}"/id_rsa
-          '' + ''
-            cat > "${cfg.dataDir}/buildkite-agent.cfg" <<EOF
-            token="$(cat ${toString cfg.tokenPath})"
-            name="${cfg.name}"
-            shell="${cfg.shell}"
-            tags="${tagStr}"
-            build-path="${cfg.dataDir}/builds"
-            plugins-path="${cfg.dataDir}/plugins"
-            hooks-path="${cfg.hooksPath}"
-            ${cfg.extraConfig}
-            EOF
-          '';
+          ## NB: maximum care is taken so that secrets (ssh keys and the CI token)
+          ##     don't end up in the Nix store.
+          preStart = let
+            sshDir = "${cfg.dataDir}/.ssh";
+            tagStr = lib.concatStringsSep "," (lib.mapAttrsToList (name: value: "${name}=${value}") cfg.tags);
+          in
+            optionalString (cfg.privateSshKeyPath != null) ''
+              mkdir -m 0700 -p "${sshDir}"
+              cp -f "${toString cfg.privateSshKeyPath}" "${sshDir}/id_rsa"
+              chmod 600 "${sshDir}"/id_rsa
+            '' + ''
+              cat > "${cfg.dataDir}/buildkite-agent.cfg" <<EOF
+              token="$(cat ${toString cfg.tokenPath})"
+              name="${cfg.name}-${toString n}"
+              shell="${cfg.shell}"
+              tags="${tagStr}"
+              build-path="${cfg.dataDir}/builds"
+              plugins-path="${cfg.dataDir}/plugins"
+              hooks-path="${cfg.hooksPath}"
+              ${cfg.extraConfig}
+              EOF
+            '';
 
-        serviceConfig =
-          { ExecStart = "${cfg.package}/bin/buildkite-agent start --config ${cfg.dataDir}/buildkite-agent.cfg";
-            User = "buildkite-agent-${name}";
-            RestartSec = 5;
-            Restart = "on-failure";
-            TimeoutSec = 10;
-            # set a long timeout to give buildkite-agent a chance to finish current builds
-            TimeoutStopSec = "2 min";
-            KillMode = "mixed";
-          };
-      };
-  });
+          serviceConfig =
+            recursiveUpdate
```

Comment:
This doesn't really need to be `recursiveUpdate` but ok

Reply:
Bad habits, I guess. I'll remove it.